File: apps/panel/web/lib/types/remote.ts
export type RemoteProtocol = 'ssh' | 'rdp';
export type RemoteCredentialScope = 'personal' | 'tenant';
export type RemoteSecretType = 'password' | 'private_key';

export type RemoteCredential = {
  id: string;
  tenantId?: string;
  accountId: string;
  name: string;
  protocol: RemoteProtocol;
  scope: RemoteCredentialScope;
  username: string;
  secretType: RemoteSecretType;
  encryptedPayload: string;
  createdAt: string;
  updatedAt: string;
  lastUsedAt?: string;
};

export type RemoteCredentialPayload = {
  name: string;
  protocol: RemoteProtocol;
  scope: RemoteCredentialScope;
  username: string;
  secretType: RemoteSecretType;
  encryptedPayload: string;
};

export type RemoteCredentialUpdatePayload = {
  name: string;
  username: string;
  secretType: RemoteSecretType;
  encryptedPayload: string;
};

export type RemoteSessionPayload = {
  accessPathId?: string;
  credentialId?: string;
  protocol: RemoteProtocol;
  username: string;
  password?: string;
  privateKey?: string;
  passphrase?: string;
  width: number;
  height: number;
  dpi: number;
};

export type RemoteSession = {
  id: string;
  token: string;
  protocol: RemoteProtocol;
  accessPathId: string;
  expiresAt: string;
  tunnelUrl: string;
};

export type RemoteAccessDefault = {
  tenantId: string;
  protocol: RemoteProtocol;
  accessPathId: string;
  updatedBy: string;
  updatedAt: string;
};

export type RemoteSecret = {
  password: string;
  privateKey: string;
  passphrase: string;
};

File: apps/panel/api/internal/domain/remote.go
package domain

const (
	RemoteProtocolSSH = "ssh"
	RemoteProtocolRDP = "rdp"

	RemoteCredentialScopePersonal = "personal"
	RemoteCredentialScopeTenant   = "tenant"
)

type RemoteCredential struct {
	ID               string `json:"id"`
	TenantID         string `json:"tenantId,omitempty"`
	AccountID        string `json:"accountId"`
	Name             string `json:"name"`
	Protocol         string `json:"protocol"`
	Scope            string `json:"scope"`
	Username         string `json:"username"`
	SecretType       string `json:"secretType"`
	EncryptedPayload string `json:"encryptedPayload"`
	CreatedAt        string `json:"createdAt"`
	UpdatedAt        string `json:"updatedAt"`
	LastUsedAt       string `json:"lastUsedAt,omitempty"`
}

type CreateRemoteCredentialInput struct {
	Name             string `json:"name"`
	Protocol         string `json:"protocol"`
	Scope            string `json:"scope"`
	Username         string `json:"username"`
	SecretType       string `json:"secretType"`
	EncryptedPayload string `json:"encryptedPayload"`
}

type UpdateRemoteCredentialInput struct {
	Name             string `json:"name"`
	Username         string `json:"username"`
	SecretType       string `json:"secretType"`
	EncryptedPayload string `json:"encryptedPayload"`
}

type RemoteSessionInput struct {
	AccessPathID string `json:"accessPathId"`
	CredentialID string `json:"credentialId"`
	Protocol     string `json:"protocol"`
	Username     string `json:"username"`
	Password     string `json:"password"`
	PrivateKey   string `json:"privateKey"`
	Passphrase   string `json:"passphrase"`
	Width        int    `json:"width"`
	Height       int    `json:"height"`
	DPI          int    `json:"dpi"`
}

type RemoteSession struct {
	ID           string `json:"id"`
	Token        string `json:"token"`
	Protocol     string `json:"protocol"`
	AccessPathID string `json:"accessPathId"`
	ExpiresAt    string `json:"expiresAt"`
	TunnelURL    string `json:"tunnelUrl"`
}

type RemoteAccessDefault struct {
	TenantID     string `json:"tenantId"`
	Protocol     string `json:"protocol"`
	AccessPathID string `json:"accessPathId"`
	UpdatedBy    string `json:"updatedBy"`
	UpdatedAt    string `json:"updatedAt"`
}

type SetRemoteAccessDefaultInput struct {
	AccessPathID string `json:"accessPathId"`
}
